src/ape/managers/config.py

```diff
@@ -183,6 +184,41 @@ def serialize(self) -> Dict:
 
         return project_config
 
+    @contextmanager
+    def using_project(self, project_folder: Path, contracts_folder: Path) -> Generator:
+        """
+        Temporarily change the project context.
+
+        Usage example::
+
+            from pathlib import Path
+            from ape import config, Project
+
+            project_path = Path("path/to/project")
+            contracts_path = project_path / "contracts"
+
+            with config.using_project(project_path, contracts_path):
+                my_project = Project(project_path)
+
+        Args:
+            project_folder (pathlib.Path): The path of the context's project.
+            contracts_folder (pathlib.Path): The path to the context's source files.
+
+        Returns:
+            Generator
+        """
+
+        initial_project_path = self.PROJECT_FOLDER
+        initial_contracts_path = self.contracts_folder
+
+        self.PROJECT_FOLDER = project_folder
+        self.contracts_folder = contracts_folder
+
+        yield
```

Comment:
curious if this might make more sense
```suggestion
        yield Project(project_folder)
```

e.g.

```py
with config.using_project(project_path, contracts_path) as project:
    ...
```

Reply:
With this, hopefully works well with the fixture:

```py
@pytest.fixture(params=project_names)
def project(request, config):
    project_source_dir = projects_directory / request.param
    project_dest_dir = config.PROJECT_FOLDER / project_source_dir.name
    copy_tree(project_source_dir.as_posix(), project_dest_dir.as_posix())
    with config.using_project(project_dest_dir, project_dest_dir / "contracts") as project:
        yield project

@pytest.fixture
def project_folder(project):
    return project.PROJECT_FOLDER
```

Reply:
I do like this idea as was able to get it to work, despite this being a cyclic dependency.